fadecandy_driver/src/fadecandy_driver.cpp

```diff
@@ -0,0 +1,300 @@
+  remaining_lookup_values.insert(remaining_lookup_values.end(), green_lookup_values.begin(), green_lookup_values.end());
+  remaining_lookup_values.insert(remaining_lookup_values.end(), blue_lookup_values.begin(), blue_lookup_values.end());
+
+  while (remaining_lookup_values.size() > 0)
+  {
+    packet.clear();
+    if (remaining_lookup_values.size() < LOOKUP_VALUES_PER_PACKET)
+    {
+      packet_lookup_values.assign(remaining_lookup_values.begin(), remaining_lookup_values.end());
+      remaining_lookup_values.erase(remaining_lookup_values.begin(), remaining_lookup_values.end());
+    }
+    else
+    {
+      packet_lookup_values.assign(remaining_lookup_values.begin(),
+                                  remaining_lookup_values.begin() + LOOKUP_VALUES_PER_PACKET);
+      remaining_lookup_values.assign(remaining_lookup_values.begin() + LOOKUP_VALUES_PER_PACKET,
+                                     remaining_lookup_values.end());
+    }
+    control = packets.size() | PACKET_TYPE_LUT;
+    if (remaining_lookup_values.size() == 0)
+    {
+      control |= FINAL_PACKET_BIT;
+    }
+
+    if (LOOKUP_VALUES_PER_PACKET - packet_lookup_values.size() > 0)
+    {
+      int j = LOOKUP_VALUES_PER_PACKET - packet_lookup_values.size();
+      for (int i = 0; i < j; i++)
+      {
+        packet_lookup_values.push_back(0);
+      }
+    }
+
+    for (size_t i = 0; i < packet_lookup_values.size(); i++)
+    {
+      int byts_per_int = 2;
+      unsigned char buffer[byts_per_int];
+      std::vector<unsigned char> bufferv;
+      bufferv = intToCharArray(buffer, packet_lookup_values[i], byts_per_int);
+      packet.insert(packet.end(), bufferv.begin(), bufferv.end());
+    }
+    // add control byte
+    packet.insert(packet.begin(), static_cast<unsigned char>(0));
+    packet.insert(packet.begin(), static_cast<unsigned char>(control));
+
+    assert(packet.size() == USB_PACKET_SIZE);
+    packets.push_back(packet);
+  }
+  return packets;
+}
+
+std::vector<int> FadecandyDriver::makeDefaultLookupTable()
+{
+  std::vector<int> lookup_values;
+  for (int row = 0; row < 257; row++)
+  {
+    lookup_values.push_back(std::min(0xFFFF, int(pow(row / 256.0, 2.2) * 0x10000)));
+  }
+  return lookup_values;
+}
+
+void FadecandyDriver::setColors(std::vector<std::vector<Color>> led_colors)
+{
+  if (fadecandy_device_ != NULL)
+  {
+    uint r = libusb_init(&context_);
+    int actual_written;
+
+    std::vector<std::vector<unsigned char>> usb_packets = FadecandyDriver::makeVideoUsbPackets(led_colors);
+    for (size_t i = 0; i < usb_packets.size(); i++)
+    {
+      r = libusb_bulk_transfer(dev_handle_, USB_ENDPOINT, usb_packets[i].data(), 64, &actual_written, 10000);
+      if (r != 0 || actual_written != 64)
+      {
+        throw std::runtime_error("Could not write on the driver.");
+      }
+    }
+  }
+  else
+  {
+    throw std::runtime_error("Device not found! Could not write on the driver.");
+  }
+}
+
+void FadecandyDriver::release()
```

Comment:
Rename the function to `releaseInterface` and you don't need the comment in your code :)